chapter2.adoc

```diff
@@ -1,214 +1,207 @@
-
-Memory and CSR accesses initiated by abstract commands or from the program buffer will be treated as if they are at the privilege level held in `prv` and `v`. These accesses will undergo protections of PMA, PMP, MMU, and other mechanisms, triggering traps if they violate corresponding rules. 
+The privilege level of the trigger chain is determined by the trigger enabled for the highest privilege level inside the chain. The entire trigger chain cannot be modified if the chain privilege level exceeds debug allowed privilege level.
 
 [NOTE]
-The external debugger has the capability to write to prv and v and subsequently read back the value, thus determining the maximum debug privilege level.  
-
-Additionally, the fields in dcsr are further constrained based on their sphere of action. For example, when a field is effective in machine mode, it is accessible only to debugger which is granted with machine mode privilege. The detailed accessibility is listed in the following table.
-
-.Dcsr fields accessibility against privilege granted to external debugger
-[options="header"]
-|============================================
-| Field    |  Allowed debug access privilege 
-| ebreakvs |  M/S/VS 
-| ebreakvu |  M/S/VS/VU 
-| ebreakm  |  M 
-| ebeaks   |  M/S 
-| ebreaku  |  M/S/U 
-| stepie   |  M 
-| stoptime |  M 
-| mprven   |  M 
-| nmip     |  M 
-|============================================
-
-===== Debug PC (dpc, at 0x7b1) and Debug Scratch Register (dscratch0, at 0x7b2; dscratch1, at 0x7b3)
-
-Debug PC (dpc) and Debug Scratch Register (dscratch0, dscratch1) are not restricted by prv and v fields to simplify the architecture.
+This represents a balance between usability and hardware complexity. There may be instances where the triggers are linked across different privilege levels (e.g., from S-mode to M-mode), while the external debugger may only have access with S-mode privilege. The external debugger should not modify the chain, because it could be suppressed or incorrectly match or fire in M-mode.
 
 ==== Sdtrig CSR
 
-The extension enforces access control in Debug Mode, which complicates trigger usage within Debug Mode. To mitigate these complications, certain trigger CSRs, tselect, tdata1, tdata2, tdata3, and tinfo are always permitted in Debug Mode, irrespective of the privileges granted to external debuggers. However, the remaining CSRs, tcontrol, scontext, hcontext, mcontext, and mscontext continue to adhere to the debug privileges granted.
+The CSRs tcontrol, scontext, hcontext, mcontext, and mscontext must follow access rules defined in <<dbgaccpriv, debug access privilege>>. Meanwhile, tselect, tdata1, tdata2, tdata3 are read/write when debug is allowed. The below table illustrates the accessibliby for Sdtrig CSRs.
 
 [options="header"]
+[cols="20%,20%,60%"]
 .Trigger CSR accessibility in Debug Mode
```

Comment:
The meaning of accessibility in the tables (Table 4-5-6) is not clear in my head. Does it meant that Writes are ignored and read return zero to prevent any information leakage.

Reply:
Yes, the fields are suppose the be R/W when it meet the condition, otherwise writes are ignored and reads return. Will rephrase it.